monai/inferers/utils.py

```diff
@@ -193,9 +197,30 @@ def sliding_window_inference(
             seg_prob_tuple = ensure_tuple(seg_prob_out)
             is_tensor_output = False
 
+        if pad_output:
+            window_shape = window_data.shape[2:]
+            seg_shape = seg_prob_tuple[0].shape[2:]
+
+            window_pad_size = []
+            window_pad_slices = []
+            for window_s, output_s in zip(window_shape, seg_shape):
+                pad_width = max(window_s - output_s, 0)
+                pad_half_1 = pad_width // 2
+                pad_half_2 = pad_width - pad_half_1
+                window_pad_size.extend([pad_half_1, pad_half_2])
+                window_pad_slices.append(slice(pad_half_1, window_s - pad_half_2))
+
+            # Make the padding area of the importance map zero
+            importance_map = torch.zeros(window_shape)
+            importance_map[window_pad_slices] = importance_map_[window_pad_slices]
+        else:
+            importance_map = importance_map_
+
         # for each output in multi-output list
         for ss, seg_prob in enumerate(seg_prob_tuple):
             seg_prob = seg_prob.to(device)  # BxCxMxNxP or BxCxMxN
+            if pad_output:
+                seg_prob = F.pad(seg_prob, pad=tuple(window_pad_size), mode="constant", value=cval)
 
             # compute zoom scale: out_roi_size/in_roi_size
```

Comment:
@drbeh thank you for updating, looks much better, but I think this is still a lot of extra code just for hovernet case here. and it's not general, what if someone wants to pad with "border" mode, or do something different. 

I suggest to only modify sliding_window_inference by adding on line 281 (just before the seg_prob_tuple for-loop)
```
if callback is not None:
    seg_prob_tuple, importance_map = callback(seg_prob_tuple, important_map)
```
and in the callback function  you can update both importance_map and seg_prob_tuple (you can define the callback from SlidingWindowHoVerNetInferer). But may be other people have a better suggestion, cc @wyli @Nic-Ma

Reply:
@myron thanks for you suggestion. I have implemented in the PR. Would be great if you can review. thanks

Reply:
@drbeh thank you so much for taking time to do these changes. to me it looks great.  I can't comment on the logic of this class for hovernet, or what this new addition is doing. but do appreciate that it does not clutter the existing class.